plugins/unicodelib/unicodelib.cpp

```diff
@@ -1475,4 +1504,22 @@ UNICODELIB_API void UNICODELIB_CALL ulUnicodeLocaleExcludeNthWord(unsigned & tgt
         tgtLen = 0;
         tgt = 0;
     }
+}
+
+UNICODELIB_API void UNICODELIB_CALL ulUnicodeLocaleTranslate(unsigned & tgtLen, UChar * & tgt, unsigned textLen, UChar const * text, unsigned searLen, UChar const * sear, unsigned replLen, UChar * repl)
+{
+    UErrorCode status = U_ZERO_ERROR;
+    UnicodeString source(text, textLen);
```

Comment:
style: source isn't the best name for a string that also returns the processed result - typically you would expect something called source to not be modified.